src/Microsoft.SqlTools.ServiceLayer/Scripting/ScripterCore.cs

```diff
@@ -268,32 +267,48 @@ internal DefinitionResult GetDefinitionUsingDeclarationType(DeclarationType type
         /// <param name="objectType">Type of SQL object</param>
         /// <returns>Location object representing URI and range of the script file</returns>
         internal Location[] GetSqlObjectDefinition(
-                ScriptGetter sqlScriptGetter,
                 string objectName,
                 string schemaName,
                 string objectType)
         {
-            StringCollection scripts = sqlScriptGetter(objectName, schemaName, null);
+            // script file destination
             string tempFileName = (schemaName != null) ? Path.Combine(this.tempPath, string.Format("{0}.{1}.sql", schemaName, objectName))
                                                 : Path.Combine(this.tempPath, string.Format("{0}.sql", objectName));
 
-            if (scripts != null)
+            ScriptingScriptOperation operation = InitScriptOperation(objectName, schemaName, objectType, tempFileName);
+            operation.Execute();
+
+            // If the scripting was successful, we should see a file with the scripts
+            bool objectFound = false;
+            int createStatementLineNumber = 0;
+            if (File.Exists(tempFileName))
             {
-                int lineNumber = 0;
-                using (StreamWriter scriptFile = new StreamWriter(File.Open(tempFileName, FileMode.Create, FileAccess.ReadWrite)))
+                using (StreamReader scriptFile = new StreamReader(tempFileName))
                 {
-
-                    foreach (string script in scripts)
+                    string line;
+                    int lineCount = 0;
+                    string createSyntax = null;
+                    if (objectScriptMap.ContainsKey(objectType))
```

Comment:
>if (objectScriptMap.ContainsKey(objectType)) [](start = 20, length = 44)

what should the else clause behavior be in this method..and please look for options to simplify the code.  perhaps though use of functions and reduce condition\looping nesting.